client/components/App.jsx

```diff
@@ -1,28 +1,46 @@
 import React from 'react'
-import {BrowserRouter as Router, Route} from 'react-router-dom'
+import {Route, withRouter} from 'react-router-dom'
 
 import '../styling/main.css'
 
 import Home from './Home'
 import Header from './Header'
 import SignIn from './SignIn'
-import Register from './Register'
 import Profile from './Profile'
+import Register from './Register'
+import WaitList from './WaitList'
 import ErrorMessage from './ErrorMessage'
 
-const App = () => {
-  return (
-    <Router>
+class App extends React.Component {
+  constructor (props) {
+    super(props)
+    this.setToken = this.setToken.bind(this)
```

Comment:
We don't need to bind `setToken`. You can remove this. Which also means the constructor is no longer needed.